### PR TITLE
Update conditions to not run certain jobs for PRs from forks

### PR DESCRIPTION
**Pull request by dhruvkb**
## Description

We observed that certain jobs fail on PRs from forks. That's because of certain permission issues. We tried to prevent these jobs from running but the `if` condition wasn't working.

This PR updates the `if` condition to something that should work to prevent these jobs from running.

## Testing Instructions

GitHub Actions are hard to test. The most we can do is a thorough code read.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -170,7 +170,7 @@ jobs:
 
   gh-pages:
     name: PR Storybook and Tailwind Preview
-    if: github.repository == 'WordPress/openverse-frontend' && github.ref != 'refs/heads/main'
+    if: github.event_name == 'pull_request' && github.event.pull_request.head.repo.owner.login == 'WordPress'
```
Comment (dhruvkb) [inline .github/workflows/ci.yml:173]: By checking that the event name is `'pull_request'`, we can skip the `ref` check.

File: .github/workflows/ghcr.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ jobs:
   push-to-ghcr:
     name: Build and push image for commit
     runs-on: ubuntu-latest
-    if: github.repository == 'WordPress/openverse-frontend'
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.owner.login == 'WordPress'
```
Comment (dhruvkb) [inline .github/workflows/ghcr.yml:15]: You can see an example of the `github.event.pull_request` object in the [GitHub PR docs](https://docs.github.com/en/rest/pulls/pulls#get-a-pull-request).

**Review by dhruvkb — COMMENTED** [review]
Some notes to help with review:

**Review by obulat — APPROVED** [review]
I used the `gh` cli to check that the `head.repo.owner.login` is indeed set to `WordPress` only for the PRs that are not made from forks.
Hope that it works 🤞

**Review by zackkrida — APPROVED** [review]
Great approach, I have a good feeling about this one

**Pull request merged**